WalletWasabi/WabiSabi/Client/CoinJoinClient.cs

```diff
@@ -315,13 +315,23 @@ private async Task<(ImmutableArray<AliceClient> aliceClientsThatSigned, IEnumera
 		try
 		{
 			var roundId = roundState.Id;
+			var roundParameters = roundState.CoinjoinState.Parameters;
 
 			registeredAliceClientAndCircuits = await ProceedWithInputRegAndConfirmAsync(smartCoins, roundState, cancellationToken).ConfigureAwait(false);
 			if (!registeredAliceClientAndCircuits.Any())
 			{
 				throw new CoinJoinClientException(CoinjoinError.NoCoinsEligibleToMix, $"The coordinator rejected all {smartCoins.Count()} inputs.");
 			}
 
+			var smallestEffectiveDenom = roundParameters.CalculateSmallestReasonableEffectiveDenomination();
+			var effectiveInputSum = registeredAliceClientAndCircuits
+				.Select(x => x.AliceClient.SmartCoin)
+				.Sum(x => x.EffectiveValue(roundParameters.MiningFeeRate));
+			if (smallestEffectiveDenom > effectiveInputSum)
+			{
+				throw new CoinJoinClientException(CoinjoinError.NoCoinsEligibleToMix, $"Aborting the round, because we couldn't register enough coins: the smallest possible denomination '{smallestEffectiveDenom}' BTC is larger than the effective input sum '{effectiveInputSum}' BTC.");
+			}
```

Comment:
The wallet handles well if this exception is thrown. CJ restarts automatically.

I wonder if we could move this check before the connection conf phase. I see no problem doing that.
Or maybe we could have it in both places? Because it's not 100% that every coin can register.

`CoinJoinClient` LOC147 could be this:
```cs
coins = CoinJoinCoinSelector.SelectCoinsForRound(coinCandidates, utxoSelectionParameters, liquidityClue);

var smallestEffectiveDenom = roundParameters.CalculateSmallestReasonableEffectiveDenomination();
var effectiveInputSum = coins
		.Sum(x => x.EffectiveValue(roundParameters.MiningFeeRate));

	if (smallestEffectiveDenom > effectiveInputSum)
	{
		throw new CoinJoinClientException(CoinjoinError.NoCoinsEligibleToMix, $"Aborting the round, because we couldn't register enough coins: the smallest possible denomination '{smallestEffectiveDenom}' BTC is larger than the effective input sum '{effectiveInputSum}' BTC.");
			}
```

Reply:
Thanks, but that's not necessary. As you may see in this PR, originally we had it implemented there, however after I realized it's better to be implemented in `SelectCoinsForRound` instead. So what's gonna now happen is that `coins` are returned empty. So what you need to confirm is if `coin` comes back empty, then we're good. This is what I tried to explain here btw: https://github.com/zkSNACKs/WalletWasabi/pull/10842#issuecomment-1578038084